helm/templates/deployment.yaml

```diff
@@ -2,35 +2,59 @@ apiVersion: extensions/v1beta1
   replicas: {{ .Values.replicas }}
   minReadySeconds: {{ .Values.minReadySeconds }}
   strategy: {}
   template:
     metadata:
       labels:
-        app.component: {{ .Values.containers.service.name }}
+        app: {{ .Values.name }}
+        environment: {{ .Values.environment }}
     spec:
       containers:   
-      - name: {{ .Values.containers.service.name }}
-        image: {{ if .Values.containers.service.registry }}{{ .Values.containers.service.registry }}/{{ end }}{{ .Values.containers.service.image }}:{{ .Values.containers.service.tag }}
-        imagePullPolicy: {{ .Values.containers.service.imagePullPolicy }}
+      - name: {{ .Values.name }}
+        image: {{ .Values.image }}
+        {{- if .Values.container.command }}
+        {{- with .Values.container.command }}
+          command:
+            {{- toYaml . | nindent 12 }}
+        {{- end }}
+        {{- with .Values.container.args }}
+          args:
+            {{- toYaml . | nindent 12 }}
+        {{- end }}
+        {{- end }}
         env:                
         - name: NODE_ENV
           value: {{ .Values.environment }}
-        - name:MESSAGE_QUEUE
-          value: {{ .Values.containers.service.messageQueue }}
-        ports:
-        - containerPort: {{ .Values.containers.service.port }}
+        - name: MESSAGE_QUEUE_HOST
+          value: {{ .Values.container.messageQueueHost }}
+        - name: MESSAGE_QUEUE_PORT
+          value: {{ .Values.container.messageQueuePort | quote}}
+        - name: MESSAGE_QUEUE_TRANSPORT
+          value: {{ .Values.container.messageQueueTransport }}
+        - name: CALCULATION_QUEUE_ADDRESS
+          value: {{ .Values.container.calculationQueueAddress }}
+        - name: CALCULATION_QUEUE_USER
+          value: {{ .Values.container.calculationQueueUser }}
+        - name: CALCULATION_QUEUE_PASSWORD
+          value: {{ .Values.container.calculationQueuePassword }}
+        - name: PAYMENT_QUEUE_ADDRESS
+          value: {{ .Values.container.paymentQueueAddress }}
+        - name: PAYMENT_QUEUE_USER
+          value: {{ .Values.container.paymentQueueUser }}
+        - name: PAYMENT_QUEUE_PASSWORD
+          value: {{ .Values.container.paymentQueuePassword }}
         resources:
           requests:
-            memory: {{ .Values.containers.service.requestMemory | quote }}
-            cpu: {{ .Values.containers.service.requestCpu | quote }}
+            memory: {{ .Values.container.requestMemory | quote }}
+            cpu: {{ .Values.container.requestCpu | quote }}
           limits:
-            memory: {{ .Values.containers.service.limitMemory | quote }}
-            cpu: {{ .Values.containers.service.limitCpu | quote }}   
+            memory: {{ .Values.container.limitMemory | quote }}
+            cpu: {{ .Values.container.limitCpu | quote }}   
         volumes:                      
-      restartPolicy: {{ .Values.containers.service.restartPolicy }}
+      restartPolicy: {{ .Values.container.restartPolicy }}
```

Comment:
There are quite a few things missing from the standard chart that will cause issues with deployments:

There are no max sure setttings
The image pull policy is not being set,
The security context settings are missing that prevent the image running as root, and make the file system read only.

Examples can be seen in the user and api-gateway services.

Reply:
I've updated the helm chart based on the API Gateway and User service.